src/pages/select-your-country.jsx

```diff
@@ -1,14 +1,35 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { graphql, navigate } from 'gatsby';
 
 import { Layout } from '../components';
 import { useCountry } from '../context/country-context';
 import Countries from '../countries';
 
+/**
+ * If the user has a country preference in session storage, we don't need to
+ * present the user to select a country. We update the button to _clear_ their selection.
+ */
 export default () => {
-  const { country, setCountry, clearCountry } = useCountry();
   const countries = Countries.getAll();
 
+  const [country, setCountry] = useState(
+    sessionStorage.getItem('country_pref')
+  );
+
+  /**
+   * When a country is selected, we store a
+   * temporary "country_pref" value in session storage
+   * that persists through reloads. This enables users to have
+   * continuity when refreshing, or returning to the site.
+   *
+   * We also bind the update to React's state, so that when a user re-selects
+   * an option, we trigger a state change to refresh the site.
+   */
+  const updateCountry = (country) => {
+    setCountry(country);
+    sessionStorage.setItem('country_pref', country);
+  };
```

Comment:
This is the core piece where we write to session storage, and update React's state.